### ISSUE TITLE
Export saved books to JSON

### ISSUE THREAD

**Issue by knjk04** [knjk04/book-project#7]
A JSON file containing a list of shelves and the books in those shelves (and its corresponding data, such as title, author and date started) should be exported to JSON.

**Comment by knjk04** [comment]
This can be unblocked books can be created and viewed from a shelf.

**Comment by knjk04** [comment]
This will be blocked until much later on. It is better to add this once all of new fields (see other issues) are added to entities.

**Comment by Tom-private** [comment]
I would like to try to implement this.

The download can be triggered using an anchor with download = true
```java
Anchor download = new Anchor(registration.getResourceUri().toString());
download.getElement().setAttribute("download", true);
download.add(new Button(new Icon(VaadinIcon.DOWNLOAD_ALT)));
```
```registration``` being a StreamRegistration for a StreamResource.

For testing I just used a dummy json, but getting json from a database should (hopefully) be well documented somewhere.

**Comment by Tom-private** [comment]
But I need more information.
Right now, I just added a new "Download" button to the top row (Imagine "Click me" as "Download"):
![image](https://user-images.githubusercontent.com/58065733/92283073-5ba78d80-eeff-11ea-948e-e5775f79a2ae.png)
Should this be the way the user triggers the download?

And which shelves should be returned?
Maybe we could read the value from the select book form and only return the selected shelf (or all shelves if nothing is selected).
Or should this serve more as a backup? So that the user always gets all data from all shelves?

**Comment by Tom-private** [comment]
New idea: When clicking the "Download" button, open a confirmation dialog, so that the user can cancel if they just clicked the button by accident / to check what it does.

Maybe it would make sense to open a WIP PR? Just describing the process may cause misunderstandings.

**Comment by knjk04** [comment]
Thanks for expressing an interest in picking up this issue! I have now assigned you to it.
>  Should this be the way the user triggers the download?

Sure, that's good.

Could you change the text of the button to 'Export' and move it into the settings page? 

> And which shelves should be returned?
Maybe we could read the value from the select book form and only return the selected shelf (or all shelves if nothing is selected).
Or should this serve more as a backup? So that the user always gets all data from all shelves?

All shelves for now. Exporting only certain shelves is a good idea, so I'll make a new issue for that.

> New idea: When clicking the "Download" button, open a confirmation dialog, so that the user can cancel if they just clicked the button by accident / to check what it does.

That's an interesting idea, but in the future, we will have a short description that explains what it does. The confirmation dialog is likely to have the same purpose (explain what clicking the button does), so I think it's a redundant step (it would be preferable from a UX perspective if a user can complete an action in the fewest number of steps).

> Maybe it would make sense to open a WIP PR? Just describing the process may cause misunderstandings.

Feel free to do that
